<sdf version='1.6'>
  <world name='default'>


    <light name='sun' type='directional'>
      <cast_shadows>1</cast_shadows>
      <pose frame=''>0 0 10 0 -0 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>

    <model name='ground_plane'>
      <static>1</static>
      <link name='link'>
        <collision name='collision'>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
          <surface>
            <contact>
              <collide_bitmask>65535</collide_bitmask>
              <ode/>
            </contact>
            <friction>
              <ode>
                <mu>100</mu>
                <mu2>50</mu2>
              </ode>
              <torsional>
                <ode/>
              </torsional>
            </friction>
            <bounce/>
          </surface>
          <max_contacts>10</max_contacts>
        </collision>
        <visual name='visual'>
          <cast_shadows>0</cast_shadows>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Grey</name>
            </script>
          </material>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
    </model>
    <gravity>0 0 -9.8</gravity>
    <magnetic_field>6e-06 2.3e-05 -4.2e-05</magnetic_field>
    <atmosphere type='adiabatic'/>
    <physics name='default_physics' default='0' type='ode'>
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>1000</real_time_update_rate>
    </physics>
    <scene>
      <ambient>0.4 0.4 0.4 1</ambient>
      <background>0.7 0.7 0.7 1</background>
      <shadows>1</shadows>
    </scene>
    <audio>
      <device>default</device>
    </audio>
    <wind/>
    <spherical_coordinates>
      <surface_model>EARTH_WGS84</surface_model>
      <latitude_deg>0</latitude_deg>
      <longitude_deg>0</longitude_deg>
      <elevation>0</elevation>
      <heading_deg>0</heading_deg>
    </spherical_coordinates>

    <model name="box">
      <pose>15 0 0.5 0 0 0</pose>
    <link name="link">
      <inertial>
        <pose>0 0 0 0 0 0</pose>
        <mass>100</mass>
        <inertia>
          <ixx>3.33341666667e+3</ixx>
          <ixy>0.0</ixy>
          <ixz>0.0</ixz>
          <iyy>3.33341666667e+3</iyy>
          <iyz>0.0</iyz>
          <izz>6.66666666667e+3</izz>
        </inertia>
      </inertial>
      <collision name="collision">
        <geometry>
          <box>
            <size>0.5 0.5 1.0</size>
          </box>
        </geometry>
        <surface>
            <friction>
              <ode>
                <mu>0</mu>
                <mu2>0</mu2>
              </ode>
            </friction>
          </surface>
      </collision>

      <visual name="visual">
        <geometry>
          <box>
            <size>0.5 0.5 1.0</size>
          </box>
        </geometry>
        <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Red</name>
            </script>
          </material>
      </visual>
    </link>
    <plugin name="plannar_mover_plugin" filename="libplannar_mover_plugin.so">
      <topic_name>test_case_topic_name</topic_name>
    </plugin>
  </model>

 
  <!-- <plugin name='gazebo_occupancy_map' filename='libgazebo_2Dmap_plugin.so'>
    <map_resolution>0.5</map_resolution> <!-- in meters, optional, default 0.1 -->
    <map_height>0.3</map_height>         <!-- in meters, optional, default 0.3 -->
    <map_size_x>200</map_size_x>          <!-- in meters, optional, default 10 -->
    <map_size_y>200</map_size_y>          <!-- in meters, optional, default 10 -->
    <init_robot_x>0</init_robot_x>          <!-- x coordinate in meters, optional, default 0 -->
    <init_robot_y>0</init_robot_y>          <!-- y coordinate in meters, optional, default 0 -->
  </plugin> -->


    <!-- <model name='ground_plane'>
        <pose frame=''>0 0 0 0 -0 0</pose>
        <scale>1 1 1</scale>
        <link name='link'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <velocity>0 0 0 0 -0 0</velocity>
          <acceleration>0 0 0 0 -0 0</acceleration>
          <wrench>0 0 0 0 -0 0</wrench>
        </link>
      </model> -->

<!-- <pose>9 -6 1.25 0 0 0</pose> FOR CROSSING-->
<!-- <pose>15 0 1.25 0 0 0</pose> FOR PASSING-->

    <!-- <actor name="actor1">
  		<pose>15 0 1.25 0 0 0</pose>
  		<skin>
    		<filename>moonwalk.dae</filename>
    		<scale>1.0</scale>
  		</skin>
      <visual name='visual'>
          <geometry>
            <cylinder>
              <radius>0.5</radius>
              <length>1</length>
            </cylinder>
          </geometry>
          <material>
            <script>
              <name>Gazebo/Grey</name>
              <uri>file://media/materials/scripts/gazebo.material</uri>
            </script>
          </material>
        </visual>
      <collision name='collision'>
          <geometry>
            <cylinder>
              <radius>0.5</radius>
              <length>1</length>
            </cylinder>
          </geometry>
          <max_contacts>10</max_contacts>
          <surface>
            <contact>
              <ode/>
            </contact>
            <bounce/>
            <friction>
              <torsional>
                <ode/>
              </torsional>
              <ode/>
            </friction>
          </surface>
        </collision>
  		<animation name="walking">
    		<filename>walk.dae</filename>
    		<scale>1.000000</scale>
    		<interpolate_x>true</interpolate_x>
  		</animation>
  		<plugin name="actor1_plugin" filename="libactor_sub.so">
    		<target>15 0 1.25</target>
    		<target_weight>1.15</target_weight>
    		<obstacle_weight>1.8</obstacle_weight>
    		<animation_factor>5.1</animation_factor>
    		<ignore_obstacles>
      			<model>ground_plane</model>
      		</ignore_obstacles>

  		</plugin>
  	</actor> -->

  </world>
</sdf>
